### PR TITLE
ssl.SSLError explicit support

### PR DESCRIPTION
**Pull request by hellysmile**
## What do these changes do?

Explicit `ssl.SSLError` support and `ssl.CertificateError`

## Are there changes in behavior for the user?

Right now `Connector` wraps any `OSError` to `ClientConnectorError `
https://github.com/aio-libs/aiohttp/blob/0b807a849564f5fefec5630b4a478789a9de1d22/aiohttp/connector.py#L813
Later `connect` wraps it one more time https://github.com/aio-libs/aiohttp/blob/0b807a849564f5fefec5630b4a478789a9de1d22/aiohttp/connector.py#L393 at this point real `OSError` is lost and 2 times `raise from`

Not sure is there any reason to catch  `OSError` in `connect`

If explicitly re raise `ClientConnectorError` only `tests.test_connector.test_connect_connection_error` fails,  but it seems mocked wrong, cuz there is no way to raise `OSError` from `_create_connection` it is already re raised to `ClientConnectorError` in `_create_direct_connection`

For first, can we remove catching `OSError` from `connect`? It makes no sense as for me

## Related issue number

https://github.com/aio-libs/aiohttp/issues/2294

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [x] Add a new news fragment into the `changes` folder
  * name it `<issue_id>.<type>` for example (588.bug)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by hellysmile** [comment]
I am not sure hot to add tests when `ssl` module is not available

File: aiohttp/connector.py [hunk 1]
```diff
@@ -390,6 +394,11 @@ def connect(self, req):
                 if self._closed:
                     proto.close()
                     raise ClientConnectionError("Connector is closed.")
+            except ClientConnectorSSLError as exc:
```
Comment (hellysmile) [inline aiohttp/connector.py]: What about just `raise` here?

**Comment by hellysmile** [comment]
What about rename `aiohttp.ClientConnectorSSLError` just to `aiohttp.ClientSSLError` ?

**Comment by fafhrd91** [comment]
will look later today

**Pull request merged**